modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectFolderResourceImpl.java

```diff
@@ -264,31 +266,41 @@ private ObjectFolder _toObjectFolder(
 						getObjectFolderItemsByObjectFolderId(
 							objectFolder.getObjectFolderId()),
 					objectFolderItem -> _toObjectFolderItem(
-						objectFolder.getObjectFolderId(), objectFolderItem),
+						objectFolder.getExternalReferenceCode(),
+						objectFolderItem),
 					ObjectFolderItem.class);
 			}
 		};
 	}
 
 	private ObjectFolderItem _toObjectFolderItem(
-			long objectFolderId,
+			String objectFolderExternalReferenceCode,
 			com.liferay.object.model.ObjectFolderItem objectFolderItem)
-		throws PortalException {
+		throws Exception {
 
 		if (objectFolderItem == null) {
 			return null;
 		}
 
-		ObjectDefinition objectDefinition =
-			_objectDefinitionLocalService.getObjectDefinition(
+		ObjectDefinitionResource.Builder builder =
+			_objectDefinitionResourceFactory.create();
+
+		ObjectDefinitionResource objectDefinitionResource = builder.user(
+			contextUser
+		).build();
+
+		com.liferay.object.admin.rest.dto.v1_0.ObjectDefinition
+			objectDefinitionDTO = objectDefinitionResource.getObjectDefinition(
```

Comment:
Since we have an inner `objectDefinition` on the ObjectFolderItem we cannot change this outer variable, but i renamed it to `dtoObjectDefinition` to fit with patterns like `serviceBuilderObjectDefinition`